bokehjs/src/lib/models/annotations/scale_bar.ts

```diff
@@ -363,67 +385,109 @@ export class ScaleBarView extends AnnotationView {
+        switch (x) {
+          case "left":   return inset.left
+          case "center": return inset.x_center
+          case "right":  return inset.right
+        }
+        const x_mapper = (() => {
+          switch (this.model.x_units) {
+            case "data":    return this.coordinates.x_scale
+            case "screen":  return panel.bbox.x_screen
+            case "view":    return panel.bbox.x_view
+            case "percent": return panel.bbox.x_percent
           }
-        case "left":
-        case "center_left":
-          return {
-            sx: hr.start + margin,
-            sy: (vr.end + vr.start)/2 - height/2,
+        })()
+        return x_mapper.compute(
+          // @ts-ignore(TS2345): Argument of type 'number | ...' is not assignable to parameter of type 'number'.
+          x,
+        )
+      })()
+      const y_pos = (() => {
+        const {y} = position
+        switch (y) {
+          case "top":    return inset.top
+          case "center": return inset.y_center
+          case "bottom": return inset.right
+        }
+        const y_mapper = (() => {
+          switch (this.model.y_units) {
+            case "data":    return this.coordinates.y_scale
+            case "screen":  return panel.bbox.y_screen
+            case "view":    return panel.bbox.y_view
+            case "percent": return panel.bbox.y_percent
           }
-        case "center":
-        case "center_center":
-          return {
-            sx: (hr.end + hr.start)/2 - width/2,
-            sy: (vr.end + vr.start)/2 - height/2,
+        })()
+        return y_mapper.compute(
+          // @ts-ignore(TS2345): Argument of type 'number | ...' is not assignable to parameter of type 'number'.
+          y,
+        )
+      })()
+      return {x: x_pos, y: y_pos}
+    })()
+
+    const anchor = (() => {
+      const anchor = resolve.anchor(this.model.anchor)
+      const x_anchor = (() => {
+        if (anchor.x == "auto") {
+          switch (position.x) {
+            case "left":   return 0.0
+            case "center": return 0.5
+            case "right":  return 1.0
+            default:       return 0.5
           }
-        case "right":
-        case "center_right":
-          return {
-            sx: hr.end - margin - width,
-            sy: (vr.end + vr.start)/2 - height/2,
+        } else {
+          return anchor.x
+        }
+      })()
+      const y_anchor = (() => {
+        if (anchor.y == "auto") {
+          switch (position.y) {
+            case "top":    return 0.0
+            case "center": return 0.5
+            case "bottom": return 1.0
+            default:       return 0.5
           }
-      }
+        } else {
+          return anchor.y
+        }
+      })()
+      return {x: x_anchor, y: y_anchor}
     })()
 
+    const width  = border_width + padding + box_layout.bbox.width  + padding + border_width
+    const height = border_width + padding + box_layout.bbox.height + padding + border_width
```

Comment:
`border_height` ?

Reply:
Border is a line thus there's only `border_width`. Similarly in CSS there's only `border-width`.